src/gui/editorwidgets/qgsdoublespinbox.cpp

```diff
@@ -142,6 +147,11 @@ void QgsDoubleSpinBox::setClearValue( double customValue, const QString &special
 
 void QgsDoubleSpinBox::setClearValueMode( QgsDoubleSpinBox::ClearValueMode mode, const QString &clearValueText )
 {
+  if ( mClearValueMode == mode && QAbstractSpinBox::specialValueText() == clearValueText )
```

Comment:
does this also need to check mCustomClearValue != 0 ?

Reply:
@nyalldawson , correct.